src/edu/hillel/lesson16/ReCalcArray.java

```diff
@@ -0,0 +1,18 @@
+package edu.hillel.lesson16;
+
+public class ReCalcArray implements Runnable{
+    private final float[] array;
+
+    public ReCalcArray(float[] array) {
+        this.array = array;
+    }
+
+    @Override
+    public void run() {
+        synchronized (array) {
```

Comment:
Do we really need synchronized block here?

Reply:
I mean, we put separated arrays and each thread is working with own part of array

Reply:
Yes, I understand that in our task we out in two threads two different arrays.
But I think that adding synchronization of this array is good - for example if somebody do such
        ReCalcArray reCalcArray1 = new ReCalcArray(arr1);
        Thread thread1= new Thread(reCalcArray1, "recalculation array 1");
        thread1.start();
        ReCalcArray reCalcArray2 = new ReCalcArray(arr1);
        Thread thread2= new Thread(reCalcArray2, "recalculation array 1");
        thread2.start();
Both threads will be recalculate arr1 - and this is not correct. That's why I added synchronized to avoid such mistake.
In ReCalcArray - we don't no who and how will create this thread. And I think I'm correct to synchronized input array.